pkg/restore/item_action.go

```diff
@@ -34,7 +34,7 @@ type ItemAction interface {
 	// should be returned, along with a warning (which will be logged but will not prevent
 	// the item from being restored) or error (which will be logged and will prevent the item
 	// from being restored) if applicable.
-	Execute(obj runtime.Unstructured, restore *api.Restore) (res runtime.Unstructured, warning error, err error)
+	Execute(obj runtime.Unstructured, fromBackup runtime.Unstructured, restore *api.Restore) (res runtime.Unstructured, warning error, err error)
```

Comment:
This has 3 inputs and 3 outputs. My PR #1125 adds a 4th output. Maybe we need to consider switching to a single input struct and an output struct + error. WDYT? cc @carlisia

Reply:
We should do this for backup item actions too, but we can do it in a separate PR.

The `warning` output variable makes things a bit messy here. We could do something like this:

```go
type RestoreItemActionExecuteOutput struct {
  UpdatedItem runtime.Unstructured
  Warning error
  Error error
}
```

but then the signature becomes `Execute(...) RestoreItemActionExecuteOutput`, with the `error` inside the struct instead of as the last output variable.

We could do this:

```go
type RestoreItemActionExecuteOutput struct {
  UpdatedItem runtime.Unstructured
  Warning error
}
```

along with `Execute(...) (RestoreItemActionExecuteOutput, error)`. Does that look weird or does it seem better?

Reply:
Indeed the `warning` makes it a little bit messy, but I can live with that. So I prefer option 2. (it will also remove warning about two errors in golint). Also, maybe others from the team would like to suggest something?
Let me know so I can make changes (but after holidays, if you prefer to do it in your PR - go ahead and I'll rebase on it)

Reply:
Added another commit with proposed changes. 
I had to tweak a little bit some tests (f.e. https://github.com/heptio/ark/pull/1123/files#diff-4cce89ebc50495fa33e9d2c373e87e6fR94) - and as I'm a golang newbie there may be a better way to do it - let me know

Reply:
Sorry for the delay. Putting the multiple args in a struct is a great way to go. I too prefer the second option.

Unfortunately the code that this code calls has all been restructured (PR #1147), but we're going to have to go thru some merging pains, all good.